### DISCUSSION TITLE
Is it possible to emit C# model classes without HTTP API client?

### DISCUSSION THREAD

**Discussion by simonthum** [microsoft/typespec#3595] [Q&A] [answered]
The use case would be to describe models, then emit JSON Schema and matching TS/C# classes, but without necessarily detailing an API. The C# classes would emit JSON compatible with the emitted Schema.

In related discussions, codegen seems to imply HTTP clients, so I figured this is a bit separate. 

The intended result would be transport agnostic interop c# types and JSON Schema.

Is that already possible, planned or out of scope?

**Comment by m-nash** [comment, marked as answer]
It is definitely possible to define only models in a tsp spec and then use the various language emitters to emit those as classes / interfaces in their respective languages without defining any services or operations in your tsp.

**Comment by simonthum** [comment]
Thanks a lot, that's what I wanted to know!
